tests/test_simple_range_query.py

```diff
@@ -0,0 +1,66 @@
+import pandas as pd
+from pandas.testing import assert_frame_equal
+
+from fsql.api import read_partitioned_table
+from fsql.query import ColumnComparator, ColumnRange, SimpleRangeQuery
+
+
+def test_simple_range_query1(tmp_path):
+    # this corresponds to c1=[b, d), c2=*, c3=[c, e)
+    ranges = [
+        ColumnRange("c1", "b", "d"),
+        ColumnRange("c2", "", "", ColumnComparator.wld),
+        ColumnRange("c3", "c", "d"),
+    ]
+    # will be omitted due to c3<b
+    left_bad1 = tmp_path / "c1=b" / "c2=a" / "c3=a"
+    # will be omitted due to c1<b
+    left_bad2 = tmp_path / "c1=a" / "c2=a" / "c3=c"
+    # will be included
+    included1 = tmp_path / "c1=c" / "c2=x" / "c3=d"
+    included2 = tmp_path / "c1=b" / "c2=4" / "c3=c"
+    # will be omitted due to c3>=e
+    right_bad1 = tmp_path / "c1=b" / "c2=w" / "c3=e"
+    # will be omitted due to c1>=d
+    right_bad2 = tmp_path / "c1=d" / "c2=w" / "c3=c"
+    for i, p in enumerate((left_bad1, left_bad2, included1, included2, right_bad1, right_bad2)):
+        p.mkdir(parents=True)
+        df = pd.DataFrame({"k": [i]})
+        df.to_csv(p / "f.csv", index=False)
+
+    query = SimpleRangeQuery(ranges=ranges)
+    result_query = read_partitioned_table(f"file://{tmp_path}/", query)
+    result_query = result_query.sort_values(by=["k"]).reset_index(drop=True)
+
+    expect = pd.DataFrame({"k": [2, 3], "c1": ["c", "b"], "c2": ["x", "4"], "c3": ["d", "c"]}).reset_index(drop=True)
+
+    assert_frame_equal(result_query, expect)
+
+
+def test_simple_range_query2(tmp_path):
+    # this corresponds to c1=[3, 27)
```

Comment:
Do we want to be left-closed, right-open? Would not be making both sides closed safer?

Reply:
intervals are usually best mixed (left-closed & right-open, or vice versa), because that is a property being preserved by union. Say you have a data query which you serve by fsql for a time range like `[yesterday midnight, today midnight)` -- then two such queries never overlap even in a single minute, so you are safe combining the results

if we had both closed, you'll overlap on the midnight-minute, if we had both open, we'll miss that minute in both

with mixed, running the two queries together is _the same_ as running a single query formed by their union. Works the other way too -- we can decide to split the range into smaller half open intervals and parallelize, for a speedup